tests/istio/tests/request_payload_limit_test.go

```diff
@@ -0,0 +1,115 @@
+		t.Log("Compass Gateway URL: ", conf.CompassGatewayURL)
+		authoizedClient := gql.NewAuthorizedHTTPClient(dexToken)
+
+		t.Log("Creating a request with big payload...")
+		bigBodyPOSTRequest, err := fixtures.FixHTTPBigBodyPOSTRequest(conf.CompassGatewayURL+directorPath, conf.RequestPayloadLimit)
+		require.NoError(t, err)
+
+		bigBodyPOSTRequest.Header.Set("Tenant", tenant)
+		bigBodyPOSTRequest = bigBodyPOSTRequest.WithContext(context.TODO())
+
+		t.Log("Executing request with big payload...")
+		resp, err := authoizedClient.Do(bigBodyPOSTRequest)
+		require.NoError(t, err)
+		defer idtokenprovider.CloseRespBody(resp)
+
+		t.Log("Executed request with big payload successfully")
+		require.Equal(t, http.StatusRequestEntityTooLarge, resp.StatusCode)
+
+		all, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Contains(t, string(all), "Payload Too Large")
+	})
+
+	t.Run("when request passes through MTLS gateway", func(t *testing.T) {
+		t.Log("Compass MTLS Gateway URL: ", conf.CompassMTLSGatewayURL)
+
+		dexGraphQLClient := gql.NewAuthorizedGraphQLClient(dexToken)
+		logrus.Infof("Registering runtime with name: %s, within tenant: %s", runtimeInput.Name, tenant)
+
+		runtime := fixtures.RegisterRuntimeFromInputWithinTenant(t, ctx, dexGraphQLClient, tenant, runtimeInput)
+		defer fixtures.UnregisterRuntime(t, ctx, dexGraphQLClient, tenant, runtime.ID)
+
+		logrus.Infof("Generating one-time token for runtime with id: %s", runtime.ID)
+		runtimeToken := fixtures.RequestOneTimeTokenForRuntime(t, ctx, dexGraphQLClient, tenant, runtime.ID)
+		oneTimeToken := &externalschema.Token{Token: runtimeToken.Token}
+
+		logrus.Info("Generating private key for cert...")
+		clientKey, err := certs.GenerateKey()
+		require.NoError(t, err)
+		tokenSecuredClient := clients.NewTokenSecuredClient(conf.CompassGatewayURL + "/connector/graphql")
+
+		logrus.Infof("Generation certificate for runtime with id: %s", runtime.ID)
+		certResult, _ := clients.GenerateRuntimeCertificate(t, oneTimeToken, tokenSecuredClient, clientKey)
+		certChain := certs.DecodeCertChain(t, certResult.CertificateChain)
+		certSecuredClient := authentication.CreateCertClient(clientKey, certChain...)
+
+		t.Log("Creating a request with big payload...")
+		bigBodyPOSTRequest, err := fixtures.FixHTTPBigBodyPOSTRequest(conf.CompassMTLSGatewayURL+directorPath, conf.RequestPayloadLimit)
+		require.NoError(t, err)
+
+		bigBodyPOSTRequest.Header.Set("Tenant", tenant)
+		bigBodyPOSTRequest = bigBodyPOSTRequest.WithContext(context.TODO())
+
+		t.Log("Executing request with big payload...")
+		resp, err := certSecuredClient.Do(bigBodyPOSTRequest)
+		require.NoError(t, err)
+		defer idtokenprovider.CloseRespBody(resp)
+
+		t.Log("Executed request with big payload successfully")
+		require.Equal(t, http.StatusRequestEntityTooLarge, resp.StatusCode)
+
+		all, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Contains(t, string(all), "Payload Too Large")
+
+	})
+}
```

Comment:
I believe that the e2e tests for the other components check that functionality. Most of them send POST requests with small payloads. We will probably have quite a few tests failing in case anything is wrong with requests having small payloads. 🙂

Reply:
Hidden dependency to other components test is not reliable. 
Best practice is always to have positive and negative tests for the newly introduced functionalities.
Having the "query" in your POST request pattern - for positive test you need to check that with random ID do not return an error message "Payload Too Large".